spikeinterface/preprocessing/remove_artifacts.py

```diff
@@ -98,22 +116,52 @@ def __init__(self, recording, list_triggers, ms_before=0.5, ms_after=3.0, mode='
         assert all(isinstance(list_triggers[i], list) for i in range(num_seg))
         assert mode in ('zeros', 'linear', 'cubic', 'average', 'median')
 
-        if ms_before is None:
-            assert ms_after is None, "To remove a single sample, set both ms_before and ms_after to None"
-        else:
+        if (ms_before is None) and (frames_before is None):
+            if (ms_after is not None) or (frames_after is not None):
+                raise ValueError("You must specify either `ms_before` and `frames_before`",
+                                 " (unless you want to remove single samples, in which case",
+                                 " set `ms_after` and `frames_after` to None)")
+            else:
+                print("Only processing samples in `list_trigger`")
+        elif (ms_before is not None) and (frames_before is None):
             ms_before = float(ms_before)
+        elif (ms_before is None) and (frames_before is not None):
+            frames_before = int(frames_before)
+        else:
+            raise ValueError("You cannot specify both `ms_before` and `frames_before`; choose one.")
+        
+        if (ms_after is None) and (frames_after is None):
+            if (ms_before is not None) or (frames_before is not None):
+                raise ValueError("You must specify either `ms_after` and `frames_after`",
+                                 " (unless you want to remove single samples, in which case",
+                                 " set `ms_before` and `frames_before` to None)")
+        elif (ms_after is not None) and (frames_after is None):
             ms_after = float(ms_after)
+        elif (ms_after is None) and (frames_after is not None):
+            frames_after = int(frames_after)
+        else:
+            raise ValueError("You cannot specify both `ms_after` and `frames_after`; choose one.")
```

Comment:
@khl02007 IMO the logic could be simpler. If `frames_before is not None`, then assert that `frames_after` is not None and use it. By default, they will be None, so if they are used they can be prioritized over the ms_before/after.